README.md
<p align="center">
  <img src="https://dotfiles.github.io/images/dotfiles-logo.png" alt="dotfiles" width="474" height="148" />
</p><Paste>

### Tools:
Bash, Neovim, Tmux, Iterm, Alacritty

### Plugin Manager
[Vim Plug](https://github.com/junegunn/vim-plug)

### NOW WITH DOCKER SUPPORT !
run `docker build . -t development`
and
`docker run -it development`


history.txt
    1  tmux
    2  xcode-select --install
    3  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
    4  ./makesymlinks.sh 
    5  cd
    6  ls -a 
    7  cd .config/
    8  ls
    9  cd
   10  cd vim/
   11  ls
   12  cd autoload/
   13  ls
   14  cd
   15  cd .config/nvim/
   16  ls
   17  vim init.vim 
   18  mv ~/Downloads/nvim-macos.tar .
   19  rat xzvf nvim-macos.tar
   20  tar xzvf nvim-macos.tar
   21  ./nvim-osx64/bin/nvim 
   22  brew install neovim
   23  vim
   24  ls
   25  rm -rf nvim-osx64/
   26  rm nvim-macos.tar 
   27  curl -fLo ~/.vim/autoload/plug.vim --create-dirs     https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
   28  sh -c 'curl -fLo "${XDG_DATA_HOME:-$HOME/.local/share}"/nvim/site/autoload/plug.vim --create-dirs \
   29         https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim'
   30  vim
   31  vim
   32  brew install pyenv
   33  exec bash
   34  brew install starship
   35  brew install openssl readline sqlite3 xz zlib
   36  git clone https://github.com/pyenv/pyenv-virtualenv.git $(pyenv root)/plugins/pyenv-virtualenv
   37  which pyenv
   38  exec bash
   39  cd
   40  cd dotfiles
   41  echo $(pyenv root)
   42  pyenv install 3.8.2
   43  brew install tmux
   44  vim a.py
   45  tmux
   46  echo $SHELL
   47  which $SHELL
   48  cd dotfiles
   49  exa
   50  brew install exa
   51  ls
   52  vim bashrc
   53  vim vimrc
   54  vim vimrc
   55  pyenv virtualenv 3.8.2 neovim3
   56  vim vimrc
   57  pip3 install neovim-remote
   58  pyenv version
   59  pyenv versions
   60  pyenv global neovim3
   61  pip install neovim-remote
   62  vim
   63  ls
   64  vim
   65  pyenv install 2.7.15
   66  pyenv virtualenv --help
   67  
   68  
   69  vim vimrc
   70  pyenv virtualenv 2.7.15 neovim2
   71  vim
   72  pyenv shell neovim2
   73  pyenv version
   74  pip install pynvim
   75  pyenv shell --help
   76  pyenv shell -
   77  vim
   78  cd
   79  cd code/
   80  ls
   81  cd ipro-item-api/
   82  pyenv virtualenv 3.8.2 itemapi
   83  pyenv local itemapi
   84  ls
   85  cd itemapi/
   86  ls
   87  vim app.py 
   88  vim ~/dotfiles/vimrc
   89  cd
   90  cd dotfiles
   91  pip freeze | grep pyls
   92  pip freeze | grep python
   93  pip list
   94  pip install -r pip.txt 
   95  pip freeze | grep python
   96  pip freeze | grep pyls
   97  pip freeze | grep auto
   98  cd
   99  cd code/ipro-item-api/
  100  vim ap
  101  vim app.py
  102  cd itemapi/
  103  vim app.py 
  104  vim ~/dotfiles/vimrc 
  105  vim app.py 
  106  git status
  107  git restore app.py 
  108  vim app.py 
  109  vim ~/dotfiles/vimrc 
  110  vim ~/.vim/LanguageClient.log 
  111  vim ~/dotfiles/vimrc 
  112  ls ~/.pyenv/shims/pyls
  113  vim ~/dotfiles/vimrc 
  114  vim app.py 
  115  vim ~/.vim/LanguageClient.log 
  116  pyls
  117  vim app.py 
  118  vim ~/dotfiles/vimrc 
  119  cd
  120  cd dotfiles
  121  vim temp.py
  122  rm temp.py 
  123  ls
  124  vim vimrc
  125  ls ~/.pyenv/versions/3.8.2/envs/neovim3/bin/python
  126  whoami
  127  pip freeze | grep python
  128  pip freeze | grep pyls
  129  pip freeze | grep pylint
  130  cd
  131  cd code/ipro-item-api/
  132  pip install python-language-server[all]
  133  cd itemapi/
  134  vim app.py 
  135  cd
  136  cd dotfiles
  137  history > history.txt


bashtheme.sh
#!/bin/sh
# base16-shell (https://github.com/chriskempson/base16-shell)
# Base16 Shell template by Chris Kempson (http://chriskempson.com)
# snow_dark scheme by nightsense (https://github.com/nightsense/snow)

color00="2c/2d/30" # Base 00 - Black
color01="be/86/8c" # Base 08 - Red
color02="7f/9d/77" # Base 0B - Green
color03="ab/91/6d" # Base 0A - Yellow
color04="75/9a/bd" # Base 0D - Blue
color05="a8/8c/b3" # Base 0E - Magenta
color06="5d/a1/9f" # Base 0C - Cyan
color07="af/b7/c0" # Base 05 - White
color08="36/3a/3e" # Base 03 - Bright Black
color09=$color01 # Base 08 - Bright Red
color10=$color02 # Base 0B - Bright Green
color11=$color03 # Base 0A - Bright Yellow
color12=$color04 # Base 0D - Bright Blue
color13=$color05 # Base 0E - Bright Magenta
color14=$color06 # Base 0C - Bright Cyan
color15="cb/d2/d9" # Base 07 - Bright White
color_foreground="af/b7/c0" # Base 05
color_background="2c/2d/30" # Base 00

if [ -n "$TMUX" ]; then
  # Tell tmux to pass the escape sequences through
  # (Source: http://permalink.gmane.org/gmane.comp.terminal-emulators.tmux.user/1324)
  put_template() { printf '\033Ptmux;\033\033]4;%d;rgb:%s\033\033\\\033\\' $@; }
  put_template_var() { printf '\033Ptmux;\033\033]%d;rgb:%s\033\033\\\033\\' $@; }
  put_template_custom() { printf '\033Ptmux;\033\033]%s%s\033\033\\\033\\' $@; }
elif [ "${TERM%%[-.]*}" = "screen" ]; then
  # GNU screen (screen, screen-256color, screen-256color-bce)
  put_template() { printf '\033P\033]4;%d;rgb:%s\007\033\\' $@; }
  put_template_var() { printf '\033P\033]%d;rgb:%s\007\033\\' $@; }
  put_template_custom() { printf '\033P\033]%s%s\007\033\\' $@; }
elif [ "${TERM%%-*}" = "linux" ]; then
  put_template() { [ $1 -lt 16 ] && printf "\e]P%x%s" $1 $(echo $2 | sed "s/\///g"); }
  put_template_var() { true; }
  put_template_custom() { true; }
else
  put_template() { printf '\033]4;%d;rgb:%s\033\\' $@; }
  put_template_var() { printf '\033]%d;rgb:%s\033\\' $@; }
  put_template_custom() { printf '\033]%s%s\033\\' $@; }
fi

# 16 color space
put_template 0  $color00
put_template 1  $color01
put_template 2  $color02
put_template 3  $color03
put_template 4  $color04
put_template 5  $color05
put_template 6  $color06
put_template 7  $color07
put_template 8  $color08
put_template 9  $color09
put_template 10 $color10
put_template 11 $color11
put_template 12 $color12
put_template 13 $color13
put_template 14 $color14
put_template 15 $color15

# foreground / background / cursor color
if [ -n "$ITERM_SESSION_ID" ]; then
  # iTerm2 proprietary escape codes
  put_template_custom Pg afb7c0 # foreground
  put_template_custom Ph 2c2d30 # background
  put_template_custom Pi afb7c0 # bold color
  put_template_custom Pj 759abd # selection color
  put_template_custom Pk 2c2d30 # selected text color
  put_template_custom Pl afb7c0 # cursor
  put_template_custom Pm 2c2d30 # cursor text
else
  put_template_var 10 $color_foreground
  if [ "$BASE16_SHELL_SET_BACKGROUND" != false ]; then
    put_template_var 11 $color_background
    if [ "${TERM%%-*}" = "rxvt" ]; then
      put_template_var 708 $color_background # internal border (rxvt)
    fi
  fi
  put_template_custom 12 ";7" # cursor (reverse video)
fi

# clean up
unset -f put_template
unset -f put_template_var
unset -f put_template_custom
unset color00
unset color01
unset color02
unset color03
unset color04
unset color05
unset color06
unset color07
unset color08
unset color09
unset color10
unset color11
unset color12
unset color13
unset color14
unset color15
unset color_foreground
unset color_background
